helix-core/src/test/java/org/apache/helix/integration/manager/TestDistributedControllerManager.java

```diff
@@ -82,8 +82,9 @@ public void simpleIntegrationTest() throws Exception {
     distributedControllers[0].disconnect();
 
     // verify leader changes to localhost_12919
-    Thread.sleep(100);
-    result = verifier.verifyByZkCallback();
+    result = TestHelper.verify(() -> {
+      return verifier.verifyByZkCallback();
+    }, 100);
```

Comment:
Actually 100, won't help much, the verify method does the following:
1. Record startTime = currentTime
2. Invoke verifier.verifyByZkCallback()
    2.1 Wait for max 30 seconds(default timeout) for the state to be verified
3.  If currentTime - startTime > 100 repeat 1.

So I think we need to have time out > (30 seconds + some buffer) if we want the verification to be retried. If we don't want the verification to be retried(which is the current implementation), then we need to use `verifier.verifyByZkCallback(30seconds default timeout + 100 milliseconds)` instead of `verifier.verifyByZkCallback()`. 

Apologies for the previous comment as I didn't read through the `TestHelper.verify` method code.

Reply:
I think, i don't need it re=tried as earlier code just did sleep and moved on. 
This is also the reason, I like to stick to the scope of the change to be what I set out with.

Reply:
Ok. Lets stick to original one or use callback with timeout that has 30 seconds + 100 milliseconds set.